### ISSUE TITLE
How does "Color By Collections" and the "Color Attribute" work

### ISSUE THREAD

**Issue by qohelet** [arangodb/arangodb#8233]
## My Environment

* __ArangoDB Version__:        3.4.1
* __Storage Engine__:             RocksDB 
* __Deployment Mode__:       Single Server 
* __Deployment Strategy__:   Manual Start in Docker
* __Operating System__:        Ubuntu 18.04
* __Disks in use__:        HDD
* __Used Package__:              Ubuntu

## Component, Query & Data

__Affected feature__:
Graph Viewer

__Problem__:
I find it quite an interesting option to have nodes in different colours, but I'm unable to find a description on how they are set properly?

**Comment by dothebart** [comment]
I'm adding the 'City Graph' from the examples Tab in `Create Graph`. 
I'm opening the 'burger menu' - switch `Add Collection Name` to `Yes`, and set `Color By Collections` to `Yes`. 
This will disable the `Color` picker, since the algorithm will auto-choose the colours of nodes. 
I know can see with purple nodes: 
- `Hambung - germanCity`, `Berlin - germanCity`, `Cologne - germanCity`. 
In contrast, I will see dark grey nodes:
- `Lyon, frenchCity`, `Paris - frenchCity`. 

If I choose `Edges` => `Add Collection Name`->`Yes` and `Color By Collections` I will see: 
- Black: `internationalHighway`
- Purple: `germanHighway`
- Blue: `frenchHighway`

does that explain it?

**Comment by qohelet** [comment]
It does explain how the "Color by Collections" attribute works, thank you.
But what is "Color Attribute" doing? Is it possible to set the color? And if so, how?

**Comment by dothebart** [comment]
you may pick `Color Attribute` instead of `Color By Collections`. 
It lets you specify an attribute on the documents they visualize to group them by color.
 I.e. `isCapital` in our case will give you:
Purple nodes for:
 - `Berlin`, `Paris`
and blue nodes for the rest.

**Comment by qohelet** [comment]
Is it possible in that case to define a colour yourself, or will it only auto-assign depending on if the values are the same or different?

**Comment by dothebart** [comment]
In the case of the color-picker only one color can be chosen, which will then be used for all nodes.

**Comment by qohelet** [comment]
But is there a way to choose the colour by yourself using an attribute from the document?

**Comment by Simran-B** [comment]
No, the color is chosen automatically if you use the Color Attribute option. Same value = same color.

**Comment by qohelet** [comment]
But dynamically from the given attributes?

**Comment by Simran-B** [comment]
The colors do not depend on the actual attribute values nor are they otherwise deterministic as far as I can tell from the code. Coloring by collection seems to take pre-defined colors ("jans") from an array in forward order. If a color is chosen by you, then this color is applied to all nodes or edges respectively (and defaults to `#2ecc71`). Color by attribute is a bit more complex, but in the end it also seems to use the *jans* colors. Relevant code:

- https://github.com/arangodb/arangodb/blob/devel/js/apps/system/_admin/aardvark/APP/frontend/js/views/graphViewer.js#L290
- https://github.com/arangodb/arangodb/blob/b1d98475ad7746cfcb0f899e73768c244c966ae8/js/apps/system/_admin/aardvark/APP/aardvark.js#L801

**Comment by OmarAyo** [comment]
Hi @qohelet 

Did you have the chance to read last Sirman's message ? Was it useful for you ? 

Can you please let us know the current status of this ticket 

Best,

**Comment by qohelet** [comment]
@OmarAyo - thank you I did, but I hoped there's some kind of chance to define the colour for each node by myself
